### PR TITLE
Lists design

### PR DESCRIPTION
**Pull request by lim-rachel**

### CODE REVIEW

File: designs/lists-design.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+### 4. Access an element, out of bounds
+*Input:*
+```
+a: [int] = None
+a = [2, 4, 6, 8]
+a[4]
+```
+*Output:*
+```
+Error: Index 4 out of bounds
```
Comment (yalhessi) [inline designs/lists-design.md]: When does this error get thrown?

**Comment by yalhessi** [comment]
Great work! I think this is a very reasonable way to approach lists. I think adding the length of the list as meta-data of a list object is a great idea, and it should prepare your implementation for future extensions (like appending to lists and printing).

File: designs/lists-design.md (hunk 1 above)
Comment (xChickens) [reply designs/lists-design.md:118]: We haven't determined this yet

Comment (xChickens) [reply designs/lists-design.md]: We're trying to throw a runtime error, which we will need to create.

**Pull request merged**